keps/sig-storage/20191125-bucket-provisioning.md

```diff
@@ -338,3 +339,170 @@ parameters: string:string [8]
 1. `secretRef`: (Optional) The name and namespace of an existing secret to be copied to the `Bucket`'s namespace for static provisioning.  Requires that `bucketIdentifier` be defined. Used for brownfield and static cases.
 1. `parameters`: (Optional) Object store specific key-value pairs passed to the driver.
 
+### RPC
+
+```
+syntax = "proto3";
+
+package cosi.v1;
+
+import "google/protobuf/descriptor.proto";
+
+option go_package = "github.com/container-object-store-interface/go-cosi";
+
+extend google.protobuf.MessageOptions {
+    bool cosi_secret = 1000;
+}
+
+message DriverInfoRequest {} // INTENTIONALLY BLANK
+
+enum DataProtocol {
+    PROTOCOL_UNSPECIFIED = 0;
+    AZURE = 1;
+    GCS = 2;
+    S3 = 3;
+}
+
+enum AccessMode {
+    MODE_UNSPECIFIED = 0;
+    RO = 1;
+    WO = 2;
+    RW = 3;
+}
+
+message DriverInfoResponse {
+    // DriverName
+    string DriverName = 1;
+
+    // SupportedProtocols
+    repeated DataProtocol SupportedProtocols = 2;
+
+    // NextId = 3;
+}
+
+message CreateBucketRequest {
+    // BucketPrefix
+    string RequestBucketPrefix = 1;
+
+    // RequestProtocol
+    DataProtocol RequestProtocol = 2;
+
+    // DriverParameters
+    map<string, string> DriverParameters = 3;
+
+    AccessMode AccessMode = 4;
+
+    // NextId = 5;
+}
+
+message AzureResponse {
+    message ConnectionData {
+        // ContainerName
+        string ContainerName = 1;
+    }
+    ConnectionData Connection = 1;
+
+    message AuthenticationData {
+        option (cosi_secret) = true;
```

Comment:
Since this spec is CO agnostic is there any worry about presuming k8s with _secret_ in this name?

Reply:
I don't think so - CSI uses `csi_secret` for the same purpose.